contents/graham_scan/code/c++/graham_scan.cpp

```diff
@@ -0,0 +1,102 @@
+#include<iostream>
+#include<vector>
+
+struct point {
+  double x;
+  double y;
+};
+
+point origin;
+
+void swap(point &p1, point &p2){
+  point temp = p1;
+  p1 = p2;
+  p2 = temp;
+}
+
+int distSq(point p1, point p2){
+  return (p1.x - p2.x)*(p1.x - p2.x) +(p1.y - p2.y)*(p1.y - p2.y);
+}
+
+int ccwCheck(point a, point b, point c){
+
+  int crossProduct = (b.y - a.y) * (c.x - b.x) - (b.x - a.x) * (c.y - b.y);
+  if (crossProduct == 0)
+  {
+    return 0;
+  }
+  return ( crossProduct > 0)? -1 : 1;
+}
+
+int compare(const void *vp1, const void *vp2){
+
+   point *p1 = (point *)vp1;
+   point *p2 = (point *)vp2;
+
+   int o = ccwCheck(origin, *p1, *p2);
+   if (o == 0)
+     return (distSq(origin, *p2) >= distSq(origin, *p1))? -1 : 1;
+
+   return (o == 1)? -1: 1;
+}
+
+void print(std::vector<point> points)
+{
+  std::cout << "the points of hull are as follows:\n";
+  for (size_t i = 0; i < points.size(); i++) {
+    std::cout << "(" << points[i].x << "," << points[i].y << ")\n";
+  }
+}
+
+std::vector<point> grahamScan(std::vector<point> points){
```

Comment:
I prefer to make this function more _c++y_ by taking the begin and end iterator as argument instead of copying the input.
 ```suggestion
 template<typename Iter>
std::vector<point> grahamScan(Iter first, Iter last){
```